### PR TITLE
# Render file creation in the refactor preview

### LINKED ISSUE #226947: Refactor Preview doesn't render newly created file content
TLDR: Refactor preview should render out the content for `createFile`/`deleteFile` not only for replace/insert/delete

Refactor preview currently doesn't render `createFile` content, instead it only renders an empty widget.

This makes it hard for users to understand what is being accepted in this case, and I would argue that the contents should be shown.
Same goes for the deleteFiles. If users don't see the original file content that is being deleted they need an extra click to jump in to the file to observe what actually is being deleted.

### PR DESCRIPTION
**Pull request by marrej**
Allows for the createFile to be rendered out as an arbitrary edit in the refactor preview

partial fix for https://github.com/microsoft/vscode/issues/226947

### CODE REVIEW

File: src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEditPreview.ts [hunk 1]
```diff
@@ -316,7 +316,11 @@ export class BulkFileOperations {
 				let ignoreAll = false;
 
 				for (const edit of file.originalEdits.values()) {
-					if (edit instanceof ResourceTextEdit) {
+					if (edit instanceof ResourceFileEdit) {
+						const content = await edit.options.contents;
```
Comment (hediet) [inline src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEditPreview.ts:320]: Can we do the awaits in parallel? (Promise.all)

**Review by hediet — COMMENTED** [review]
Thanks, looks good! Left a comment.

File: src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEditPreview.ts (hunk 1 above)
Comment (marrej) [reply src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEditPreview.ts:320]: Fair point. PTAL

**Review by hediet — APPROVED** [review]

**Review by hediet — APPROVED** [review]
Thanks!

**Review by lszomoru — APPROVED** [review]

**Pull request merged**